raitools/raitools/_managers/explainer_manager.py

```diff
@@ -0,0 +1,135 @@
+from raitools._managers.base_manager import BaseManager
+
+SparseNumFeaturesThreshold = 1000
+
+
+class ExplainerManager(BaseManager):
+
+    """Defines the ExplainerManager for explaining a model.
+
+    :param model: The model to explain.
+        A model that implements sklearn.predict or sklearn.predict_proba
+        or function that accepts a 2d ndarray.
+    :type model: object
+    :param initialization_examples: A matrix of feature vector
+        examples (# examples x # features) for initializing the explainer.
+    :type initialization_examples: pandas.DataFrame
+    :param evaluation_examples: A matrix of feature vector
+        examples (# examples x # features) on which to explain the
+        model's output.
+    :type evaluation_examples: pandas.DataFrame
+    :param target_column: The name of the label column.
+    :type target_column: str
+    :param classes: Class names as a list of strings.
+        The order of the class names should match that of the model
+        output.  Only required if explaining classifier.
+    :type classes: list
+    """
+
+    def __init__(self, model, initialization_examples, evaluation_examples,
+                 target_column, classes=None):
+        """Defines the ExplainerManager for explaining a model.
+
+        :param model: The model to explain.
+            A model that implements sklearn.predict or sklearn.predict_proba
+            or function that accepts a 2d ndarray.
+        :type model: object
+        :param initialization_examples: A matrix of feature vector
+            examples (# examples x # features) for initializing the explainer.
+        :type initialization_examples: pandas.DataFrame
+        :param evaluation_examples: A matrix of feature vector
+            examples (# examples x # features) on which to explain the
+            model's output.
+        :type evaluation_examples: pandas.DataFrame
+        :param target_column: The name of the label column.
+        :type target_column: str
+        :param classes: Class names as a list of strings.
+            The order of the class names should match that of the model
+            output.  Only required if explaining classifier.
+        :type classes: list
+        """
+        self._model = model
+        self._initialization_examples = \
+            initialization_examples.drop(columns=[target_column])
+        self._evaluation_examples = \
+            evaluation_examples.drop(columns=[target_column])
+        self._is_run = False
+        self._is_added = False
+        self._surrogate_model = None
+        self._features = list(self._initialization_examples.columns)
+        self._classes = classes
+        self._target_column = target_column
+        self._explanation = None
+
+    def add(self):
+        """Add an explainer to be computed later."""
+        if self._is_added:
+            warnings.warn(("Ignoring.  Explanation has already been added, "
```

Comment:
Shouldn't a user be able to add as many explanations as they want?

Reply:
I don't think it makes sense to add multiple explanations, since they are all computed on the same model/dataset, but it will make sense for other things since they have parameters on the add method

Reply:
Might a user not want to have both Mimic and Shap explanations for a Model? OK, we're not allowing that now, but we should have the machinery set up to do that.

Reply:
yes, we can change that machinery later, but right now (and possibly in the future) I want to keep explainers as simple as possible in this new API

Reply:
I think @minthigpen had this in her plans already. Can you confirm whether this is a short-term feature to add or not @minthigpen ?

Reply:
I think we agreed in several meetings that we would only support a single explainer for now, I definitely don't want to support all of them for the initial release, or even after that, it requires a lot of maintenance.  At that point, it would be much better to allow the user to pass in the computed explanation object instead of the explainer.

Reply:
Certainly only support one particular explainer in what's shown to the user, but why not build the machinery for more now? We know that the requirement for having more than one explanation is coming, so why not build the infrastructure now?

Reply:
I guess I don't agree with that requirement when it comes to explainers, I'm not sure we will want to support more than one explainer.  Or, if we will, then we should just take in explanations, we shouldn't have even more wrappers around our explainers, we already have too many wrappers to maintain.

Reply:
Users will never want to try explaining their model with more than one technique?

Reply:
If they will, RAIAnalyzer should take in the explanations, not explainers.  Or if they want to customize their explanations this much they should use the ExplanationDashboard instead.  Trying to maintain all of the wrappers for everything will create a lot of bugs and will be a maintenance headache.

Reply:
I may have been mixing up explanations and explainers and creating confusion that way.

Taking a step back, I'm thinking of a case where (once we've allowed the user to select whether they want to use LIME/shap/Mimic etc.) the user looks at the 'Explanation' tab of the dashboard and says "This isn't showing me what I need." Or perhaps "This shows me what I need to know in region X of parameter space, but not in region Y." They will then want to generate another interpretation of their model to rectify this, but without regenerating the entire `RAIAnalyzer` object - after all, they may well have been using other components and not want to be forced to recompute all of those.

In this case, they're going to end up with two interpretations hanging off the `ExplainerManager`.

Does that make more sense?

Reply:
Yes, we discussed offline, agree that this is something we might want to plan/discuss more, but currently it's not needed and out of scope for the initial implementation.  Actually my hope is this API can somehow take an explanation instead or we just keep the simple default explainer, since adding every explainer/parameter will be nontrivial and could also create a lot of issues/bugs/confusion since we would then have to maintain those wrappers.  But for now based on our offline discussion we seemed to agree that it's fine to punt this until later, as changing this implementation shouldn't be too hard and shouldn't break backwards compatibility since we will always have a `default` case.